### PR TITLE
Update changelog for `@metamask/gas-fee-controller@6`

### PR DESCRIPTION
**Pull request by Gudahtt**
## Description

The changelog has been updated to better describe the changes. A new entry has been added for a newly discovered breaking change, and an existing entry was updated to clarify when it applies.

## Changes

None

## References

Relates to changes in #1329

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]

File: packages/gas-fee-controller/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - **BREAKING:** The `getChainId` constructor parameter now expects a `Hex` return type rather than a decimal string ([#1367](https://github.com/MetaMask/core/pull/1367))
 - Add `@metamask/utils` dependency
 - **BREAKING:** The gas fee controller messenger now requires the `NetworkController:stateChange` event instead of the `NetworkController:providerConfigChange` event ([#1329](https://github.com/MetaMask/core/pull/1329))
+  - This does not apply if `onNetworkStateChange` and `getChainId` are provided to the constructor
+- **BREAKING:** The constructor parameter `onNetworkStateChange` now expects event hanlders to be passed the full network state ([#1329](https://github.com/MetaMask/core/pull/1329))
```
Comment (mcmire) [inline packages/gas-fee-controller/CHANGELOG.md:16]: Typo:

``` suggestion
- **BREAKING:** The constructor parameter `onNetworkStateChange` now expects event handlers to be passed the full network state ([#1329](https://github.com/MetaMask/core/pull/1329))

**Review by mcmire — COMMENTED** [review]
Just one thing.

**Review by mcmire — APPROVED** [review]
Looks good!

**Comment by mcmire** [comment]
@Gudahtt Should I edit the [GitHub release](https://github.com/MetaMask/core/releases/tag/v53.0.0) to include these changes as well?

**Pull request merged**